stdlib/bytes.gr

```diff
@@ -0,0 +1,549 @@
+
+import Memory from "runtime/unsafe/memory"
+import WasmI32 from "runtime/unsafe/wasmi32"
+import WasmI64 from "runtime/unsafe/wasmi64"
+import WasmF32 from "runtime/unsafe/wasmf32"
+import WasmF64 from "runtime/unsafe/wasmf64"
+import {
+  tagSimpleNumber,
+  allocateBytes,
+  newInt32,
+  newFloat32,
+  newInt64,
+  newFloat64
+} from "runtime/dataStructures"
+import { coerceNumberToInt32, coerceNumberToInt64 } from "runtime/numbers"
+import Exception from "runtime/exception"
+
+// -----------------------------------------------------------------------------
+// Conversion Utils
+// -----------------------------------------------------------------------------
+
+/** Converts Number to WasmI32 */
+@disableGC
+let numberToWasmI32 = n => {
+  let i32 = coerceNumberToInt32(n);
+  let ptr = WasmI32.fromGrain(i32);
+  WasmI32.load(ptr, 8n)
+}
+
+@disableGC
+let int32ToWasmI32 = (n: Int32) => {
+  WasmI32.load(WasmI32.fromGrain(n), 8n)
+}
+
+@disableGC
+let float32ToWasmF32 = (n: Float32) => {
+  WasmF32.load(WasmI32.fromGrain(n), 8n)
+}
+
+@disableGC
+let int64ToWasmI64 = (n: Int64) => {
+  WasmI64.load(WasmI32.fromGrain(n), 8n)
+}
+
+@disableGC
+let float64ToWasmF64 = (n: Float64) => {
+  WasmF64.load(WasmI32.fromGrain(n), 8n)
+}
+
+@disableGC
+let wasmI32ToInt32 = (n: WasmI32) => {
+  WasmI32.toGrain(newInt32(n)) : Int32
+}
+
+@disableGC
+let wasmI32ToFloat32 = (n: WasmF32) => {
+  WasmI32.toGrain(newFloat32(n)) : Float32
+}
+
+@disableGC
+let wasmI64ToInt64 = (n: WasmI64) => {
+  WasmI32.toGrain(newInt64(n)) : Int64
+}
+
+@disableGC
+let wasmF64ToFloat64 = (n: WasmF64) => {
+  WasmI32.toGrain(newFloat64(n)) : Float64
+}
+
```

Comment:
@ospencer I can pretty much remove all these conversion utils if we can land https://github.com/grain-lang/grain/pull/656 or find some other solution 🎉